openshift-ci/deploy_metallb.sh

```diff
@@ -3,8 +3,19 @@
 metallb_dir="$(dirname $(readlink -f $0))"
 source ${metallb_dir}/common.sh
 
+# Extract and format the oc version to branch
+extract_version() {
+  local output="$1"
+  local version_line=$(echo "$output" | grep "$2")
+  local version=$(echo "$version_line" | awk '{print $3}')
+  local major=$(echo "$version" | cut -d '.' -f 1)
+  local minor=$(echo "$version" | cut -d '.' -f 2)
+  local oc_branch="release-$major.$minor"
+  echo "$oc_branch"
+}
+OC_BRANCH=$(extract_version "$(oc version)" "Server Version:")
+git checkout ${OC_BRANCH}
```

Comment:
Why do we need to checkout here?

Reply:
Because currently we are checking out metallb into METALLB_BRANCH from the release ci script (and then invoking this script inside metallb)

Reply:
I think the logic of branch selection must happen in the ci script.
This may work if we call the CI from the metallb repo (the metallb branch is the right one, but we need to be instructed about the operator one), but if the CI is called either from FRR or from the operator, the checkout of the metallb repo happens here https://github.com/openshift/release/blob/a2a4604ee4448bfec4c9084c4caf575f00f57583/ci-operator/step-registry/baremetalds/metallb-e2e/test/baremetalds-metallb-e2e-test-commands.sh#L33

Reply:
trying here:
https://github.com/openshift/release/pull/41419